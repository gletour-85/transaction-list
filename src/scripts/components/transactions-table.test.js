import React from 'react';
import renderer from 'react-test-renderer';
import TransactionsTable from './transactions-table';

import mocked_useTransactions from '../hooks/use-transactions';

// Mocks
// -----

jest.mock('@material-ui/core', () => ({
  CircularProgress: 'CircularProgress',
  Paper: 'Paper',
  Table: 'Table',
  TableBody: 'TableBody',
  TableCell: 'TableCell',
  TableContainer: 'TableContainer',
  TableHead: 'TableHead',
  TableRow: 'TableRow'
}));
jest.mock('../hooks/use-transactions');

// Setup
// -----

function getComponent() {
  return renderer.create(<TransactionsTable />);
}

// Tests
// -----

describe('TransactionsTable', () => {
  let component;

  describe('rendering', () => {
    describe('when transactions are loading', () => {
      beforeEach(() => {
        mocked_useTransactions.mockReturnValue({
          isLoading: true
        });

        component = getComponent();
      });

      it('renders correctly', () => {
        expect(component).toMatchSnapshot();
      });
    });

    describe('when transactions failed to load', () => {
      beforeEach(() => {
        mocked_useTransactions.mockReturnValue({
          error: { message: 'request has timed out' },
          isLoading: false
        });

        component = getComponent();
      });

      it('renders correctly', () => {
        expect(component).toMatchSnapshot();
      });
    });

    describe('when transactions have loaded', () => {
      beforeEach(() => {
        mocked_useTransactions.mockReturnValue({
          isLoading: false,
          transactions: [
            { Date: '2021-05-23', Company: 'mock-company-1', Ledger: 'mock-debit-1', Amount: '-5.00' },
            { Date: '2021-05-24', Company: 'mock-company-2', Ledger: 'mock-debit-2', Amount: '-10.25' },
            { Date: '2021-05-25', Company: 'mock-company-3', Ledger: 'mock-credit-1', Amount: '5.00' },
            { Date: '2021-05-26', Company: 'mock-company-4', Ledger: 'mock-credit-2', Amount: '20.50' }
          ]
        });

        component = getComponent();
      });

      it('renders correctly', () => {
        expect(component).toMatchSnapshot();
      });
    });
  });
});
