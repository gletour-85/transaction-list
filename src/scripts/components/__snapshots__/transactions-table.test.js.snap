// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionsTable rendering when transactions are loading renders correctly 1`] = `
<CircularProgress
  className="transactions-table--loading"
  size={100}
/>
`;

exports[`TransactionsTable rendering when transactions failed to load renders correctly 1`] = `
<div>
  Unable to load transactions, error: request has timed out
</div>
`;

exports[`TransactionsTable rendering when transactions have loaded renders correctly 1`] = `
<TableContainer
  component="Paper"
>
  <Table
    aria-label="transactions table"
    className="transactions-table"
  >
    <TableHead
      className="transactions-table__header"
    >
      <TableRow>
        <TableCell>
          Date
        </TableCell>
        <TableCell>
          Company
        </TableCell>
        <TableCell>
          Account
        </TableCell>
        <TableCell>
          $10.25
        </TableCell>
      </TableRow>
    </TableHead>
    <TableBody
      className="transactions-table__body"
    >
      <TableRow>
        <TableCell
          className="transactions-table__muted-cell"
        >
          2021-05-23
        </TableCell>
        <TableCell>
          mock-company-1
        </TableCell>
        <TableCell
          className="transactions-table__muted-cell"
        >
          mock-debit-1
        </TableCell>
        <TableCell
          className=""
        >
          -5.00
        </TableCell>
      </TableRow>
      <TableRow>
        <TableCell
          className="transactions-table__muted-cell"
        >
          2021-05-24
        </TableCell>
        <TableCell>
          mock-company-2
        </TableCell>
        <TableCell
          className="transactions-table__muted-cell"
        >
          mock-debit-2
        </TableCell>
        <TableCell
          className=""
        >
          -10.25
        </TableCell>
      </TableRow>
      <TableRow>
        <TableCell
          className="transactions-table__muted-cell"
        >
          2021-05-25
        </TableCell>
        <TableCell>
          mock-company-3
        </TableCell>
        <TableCell
          className="transactions-table__muted-cell"
        >
          mock-credit-1
        </TableCell>
        <TableCell
          className="transactions-table__credit-cell"
        >
          5.00
        </TableCell>
      </TableRow>
      <TableRow>
        <TableCell
          className="transactions-table__muted-cell"
        >
          2021-05-26
        </TableCell>
        <TableCell>
          mock-company-4
        </TableCell>
        <TableCell
          className="transactions-table__muted-cell"
        >
          mock-credit-2
        </TableCell>
        <TableCell
          className="transactions-table__credit-cell"
        >
          20.50
        </TableCell>
      </TableRow>
    </TableBody>
  </Table>
</TableContainer>
`;
